src/portable/portabled.c

```diff
@@ -43,6 +44,14 @@ static int manager_new(Manager **ret) {
         if (r < 0)
                 return r;
 
+        r = sd_event_add_signal(m->event, NULL, SIGRTMIN+18, sigrtmin18_handler, NULL);
+        if (r < 0)
+                return r;
+
+        r = sd_event_add_memory_pressure(m->event, NULL, NULL, NULL);
```

Comment:
Does this make sense on a run-to-completion activatable service? Isn't it better to just complete the tasks as quickly as possible and exit? Maybe shorten the idle timer when pressure is detected instead?

Reply:
Oh, there's no doubt, the actions to take on memory pressure are not polished yet. for example udevd/userdbd should stop idle worker processes on memory pressure, and services that safely can exit-on-idle should do so immediately. We should add more of this, no doubt. But adding that usually requires some more substantial rework.

In contrast this one-liner above is very easy to add and already very effective. It's also dead cheap for tiny process (like portabled). There's really no risk in calling that on memory pressure.